### DISCUSSION TITLE
Supporting additional RF sensors

### DISCUSSION THREAD

**Discussion by JohnOH** [HeyuX10Automation/heyu#68] [Q&A]
Hello Janusz,

Long time no speak. I am still running Heyu sucessfully with many thanks to you. I recently spotted a 433Mhz temperature/humidity sensor that is decoded by my RFXCOM. I wondered if this might be plumbed in to my Heyu config file and handled by the X10 extension or some such. 
The device can transmit in three modes although 1&3 appear to be the same.

Switch positions 1 & 3 Temperature & Humidity:
17/08/2023 06:05:55:194= 0A520A0982F901093701A9
Packettype    = TEMP_HUM
subtype       = TH10 - Rubicson,IW008T,TX95,Xiron-EN6,WH5
                channel 249
Sequence nbr  = 9
ID            = 82F9 decimal:33529
Temperature   = 26.5 °C
Humidity      = 55
Status        = Comfortable
Signal level  = 10  -40dBm
Battery       = OK

Switch postion 2 Temperature only:
17/08/2023 06:04:43:665= 085009088601010A99
Packettype    = TEMP
subtype       = TEMP9 - RUBiCSON 48695
Sequence nbr  = 8
ID            = 8601 decimal:34305
Temperature   = 26.6 °C
Signal level  = 9  -48dBm
Battery       = OK

Perhaps the Heyu code is structured in a way that might permit me to add the device.

Many thanks, John.

**Comment by jkrzyszt** [comment]
Hello @JohnOH,

I'm sorry for no response.  In short, I have no resources to work on new Heyu features.

In regard to your Rubiscon sensor, I found an open source project called rtl_433 that supports a lot of sensors, including Rubiscon.  AFAICT, rtl_433 packages are available in Debian / Ubuntu.  That software doesn't support RFXCOM, it requires a cheap Realtek RTL2832 based DVB dongle.  It supports a number of output formats/protocols, including MQTT, but I've never used it so I'm not sure how to integrate it with Heyu, if that's your requirement.

I'm still running Heyu, but limited to controlling X10 powerline devices.  I have disconnected my RFXCOM RFXLAN from Heyu and connected it to Home Assistant.  My RFXLAN runs RFXCOM xPL firmware and talks to Home Assistant over my custom software adapter implemented with Node-red.

I don't think Heyu code is structured in a way that is easily extendable with new devices, but that's still possible.  If you would like to keep your project based on Heyu then I think you could read the rtl_433 code and reimplement Rubiscon support in Heyu, with credits to original authors.
